compiler/src/dotty/tools/dotc/printing/Formatting.scala

```diff
@@ -77,17 +77,16 @@ object Formatting {
   class SyntaxFormatter(sc: StringContext) extends StringFormatter(sc) {
     override protected def showArg(arg: Any)(implicit ctx: Context): String =
       arg match {
-        case arg: Showable if ctx.settings.color.value != "never" =>
-          val highlighted =
-            SyntaxHighlighting(wrapNonSensical(arg, super.showArg(arg)))
-          new String(highlighted.toArray)
         case hl: Highlight =>
           hl.show
         case hb: HighlightBuffer =>
           hb.toString
         case str: String if ctx.settings.color.value != "never" =>
           new String(SyntaxHighlighting(str).toArray)
-        case _ => super.showArg(arg)
+        case x :: xs =>
+          showArg(x) + (if (xs.isEmpty) "" else ", " + showArg(xs))
```

Comment:
I would remove this special case. This seems completely arbitrary to special case `List`.
If a user want to pretty print a `List` like this, then he should do it himself.